crates/csharp/src/function.rs

```diff
@@ -956,11 +1003,24 @@ impl Bindgen for FunctionBindgen<'_, '_> {
             }
 
             Instruction::CallWasm { sig, .. } => {
+                let is_async = InterfaceGenerator::is_async(self.kind);
+
+                let requires_async_return_buffer_param = is_async && sig.results.len() >= 1;
+                let async_return_buffer = if requires_async_return_buffer_param {
+                    let buffer = self.emit_allocation_for_type(&sig.results);
+                    uwriteln!(self.src, "//TODO: store somewhere with the TaskCompletionSource.");
```

Comment:
I think you can use `new TaskCompletionSource(object? state)` -> `Task.AsyncState`